src/cmd/input.go

```diff
@@ -0,0 +1,66 @@
+package cmd
+
+import (
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var dataFile string
+
+func hasStdin() bool {
+	stat, err := os.Stdin.Stat()
+	if err != nil {
+		return false
+	}
+	return stat.Size() > 0
+}
+
+func readInputFile() ([]byte, error) {
+	if hasStdin() {
+		data, err := io.ReadAll(os.Stdin)
+		if err != nil {
+			return nil, err
+		}
+		return data, nil
+	} else {
+		file, err := os.Open(dataFile)
+		if err != nil {
+			return nil, err
+		}
+		defer file.Close()
+
+		data, err := io.ReadAll(file)
+		if err != nil {
+			return nil, err
+		}
+		return data, nil
+	}
+}
+
+func readInputConfig() {
```

Comment:
I named this function `InputFile` rather than `InputConfigFile` so it's easier to differentiate with `readInputFile`.

This handles viper config, the latter handles an actual file (or stdin)